libs/containers/include/mrpt/containers/vector_with_small_size_optimization.h

```diff
@@ -161,15 +161,14 @@ class vector_with_small_size_optimization
 
 	const_reference back() const
 	{
-		return m_is_small ? m_a.back() : m_v.back();
+		return m_is_small ? m_a[m_size - 1] : m_v.back();
```

Comment:
I had to change this to fix a regression in tests for CVectorDouble and push_back()... if you think about it a bit it makes sense after all. It's undefined for empty containers, just like the std.

So good the body of unit tests grows.

Reply:
I was just thinking about the trivial case where `m_size = 0` then `front() == back()`. `front()` should be m_a[0] always. I think you want `m_a[m_size]`, not `m_a[m_size - 1]` because `m_a[m_size]` is `m_a[0]` when the array is empty. Did I miss something?

Reply:
> front() should be m_a[0] always.
Sure!

> I think you want m_a[m_size], not m_a[m_size - 1]
Hmm... that would be where `end()` points to, but `back()` is the ref to the last valid element... isn't it?
I agree in that the trivial case with `m_size==0` would lead to `size_t(-1)`, and I wasn't comfortable with that, until I saw that for STL containers, it's undefined behavior to call `back()` when empty, so I saw it was acceptable to just... leave it "undefined behavior" (actually, would be a crash). 

We could add an ASSERT_() for debug builds, on a second thought...

Reply:
Right. I was thinking of end(). Thx. I got it now.